evmd/cmd/evmd/cmd/root.go

```diff
@@ -315,10 +316,19 @@ func newApp(
 
 	// Set up the required mempool and ABCI proposal handlers for Cosmos EVM
 	baseappOptions = append(baseappOptions, func(app *baseapp.BaseApp) {
-		mempool := sdkmempool.NoOpMempool{}
-		app.SetMempool(mempool)
-
-		handler := baseapp.NewDefaultProposalHandler(mempool, app)
+		var mpool mempool.Mempool
+		if maxTxs := cast.ToInt(appOpts.Get(server.FlagMempoolMaxTxs)); maxTxs >= 0 {
```

Comment:
I'm testing this with a Foundry script that uses transaction batching on this branch, and it seems like this is still failing to process out-of-order nonces.

Getting these errors:
```
- server returned an error response: error code -32000: invalid nonce; got 24, expected 23: invalid sequence: invalid sequence
```

Afaik, this PR is supposed to somewhat mitigate that. Is that not the case, or am I doing something wrong here?

Reply:
Do you mind share the script? but we use different [senders](https://github.com/MANTRA-Chain/mantrachain-e2e/blob/d4bc921dce85f69e8aa060e6d844d0ba67a636eb/integration_tests/test_priority.py#L53) for priority test.

Reply:
Got it, since need specify nonce from same sender to avoid out of order in priority mempool, I could try sender nonce mempool later.

Reply:
`PriorityNonceMempool` don't handle out of order nonce, it only handles priority.